src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1243,6 +1247,35 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                             break;
                         }
 
+                    case DeclarationModifiers.File:
+                        // 'file' is only a modifier if it is followed by an optional sequence of modifiers and then by a type keyword.
+                        {
+                            for (int i = 1; ; i++)
+                            {
+                                var kind = PeekToken(i).ContextualKind;
+                                if (kind is SyntaxKind.ClassKeyword
+                                        or SyntaxKind.StructKeyword
+                                        or SyntaxKind.InterfaceKeyword
+                                        or SyntaxKind.RecordKeyword
+                                        or SyntaxKind.EnumKeyword
+                                        or SyntaxKind.DelegateKeyword)
+                                {
+                                    // this is a file modifier.
+                                    modTok = ConvertToKeyword(EatToken());
+                                    break;
+                                }
+
+                                if (SyntaxFacts.IsKeywordKind(kind))
+                                {
+                                    continue;
+                                }
+
+                                return;
+                            }
```

Comment:
Can this somehow overflow for an error scenario in a large file?
What if `file` isn't followed by any of the things we're looking for. Will `i` passed to `PeekToken(i)` get out of range? I'm not sure what `PeekToken` will do in that case, or if this case can even happen?

Reply:
I think we expect to get an EOF token. But it's a good case to test, e.g. `file` by itself or `file async` by itself.